src/nfc.android.ts

```diff
@@ -419,6 +409,29 @@ export class Nfc implements NfcApi {
     });
   }
 
+  private initNfcAdapter() {
+    if (!this.created) {
+      const activity = application.android.foregroundActivity || application.android.startActivity;
+      if (activity) {
+        this.created = true;
+        this.intent = new android.content.Intent(activity, activity.getClass());
+        this.intent.addFlags(android.content.Intent.FLAG_ACTIVITY_SINGLE_TOP | android.content.Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        this.pendingIntent = android.app.PendingIntent.getActivity(activity, 0, this.intent, 0);
+
+        // The adapter must be started with the foreground activity.
+        // This allows to start it as soon as possible but only once.
+        const foregroundActivity = application.android.foregroundActivity;
```

Comment:
On line 414 `activity` *may* be unset and instead resolve to `application.android.foregroundActivity`, so the `foregroundActivity` will be unset in that case. Would it help to change line 414 to `const activity = application.android.foregroundActivity` instead?

Reply:
Ah, now I see there's a check `foregroundActivity !== null` on line 425. Maybe that should be pulled up, but I guess this works fine nonetheless.

Reply:
Yes, the bottom line of this pr is that I reorganized the existing code so that we use an explicit foreground activity when we need it, and only if it's ready. But we still use either a foreground activity or a start activity when it doesn't matter.
Appart from that and the use of a private method (to avoid repetition), there is hardly anything new :)

I removed the setTimeout and try to create the handler immediately (which may work if we import the plugin after a foreground activity already exists, like inside of a view-model). I also try again after the start activity event fires in case the handler is not yet  created (which is important if we import the plugin at the start of the app, and no activity exists yet)

Reply:
Yes, the bottom line of this pr is that I reorganized the existing code so that we use an explicit foreground activity when we need it, and only if it's ready. But we still use either a foreground activity or a start activity when it doesn't matter.
Appart from that and the use of a private method (to avoid repetition), there is hardly anything new :)

I removed the setTimeout and try to create the handler immediately (which may work if we import the plugin after a foreground activity already exists, like inside of a view-model). I also try again after the start activity event fires in case the handler is not yet created (which is important if we import the plugin at the start of the app, and no activity exists yet)

Reply:
I like it, thank you!

Reply:
My pleasure!
Do you know when the 4.0.4 may be released?

Reply:
I hope today even!